controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,17 +105,15 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to upgrade kubelet config map")
 	}
 
-	status, err := workloadCluster.ClusterStatus(ctx)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
+	cMachineList := controlPlane.Machines
+	machineCount := int32(cMachineList.Len())
```

Comment:
nit avoid the local vars given that those values are used in a single point only